crates/core/tedge/src/cli/certificate/cli.rs

```diff
@@ -44,6 +57,18 @@ impl BuildCommand for TEdgeCertCli {
                     id,
                     cert_path: config.device.cert_path.clone(),
                     key_path: config.device.key_path.clone(),
+                    csr_path: None,
+                };
+                cmd.into_boxed()
+            }
+
+            TEdgeCertCli::CreateCsr { id, output_path } => {
+                let cmd = CreateCsrCmd {
+                    id,
+                    cert_path: config.device.cert_path.clone(),
+                    key_path: config.device.key_path.clone(),
+                    csr_path: config.device.csr_path.clone(),
+                    output_path,
```

Comment:
This is where the configured csr path as a default when no output path is provided on the cli.


```suggestion
                    // Use output file instead of csr_path from tedge config if provided
                    csr_path: output_path.unwrap_or_else(|| config.device.csr_path.clone()),
```